### ISSUE TITLE
ctx command: display context even when context-sections is empty?

### ISSUE THREAD

**Issue by disconnect3d** [pwndbg/pwndbg#2510]
This is to be discussed: I find it annoying that the `ctx` command displays nothing when `set context-sections ''` is empty.

It should probably display the default context at this point?

**Comment by disconnect3d** [comment]
@gsingh93 @CptGibbon @Aplet123 @OBarronCS @patryk4815 asking for input here, wdyt? :)

**Comment by CptGibbon** [comment]
Does `set context-sections ''` have a legitimate use case?

**Comment by disconnect3d** [comment]
Yeah, I use it very often when scripting/debugging something extensively - I set up `dprintf free, "free(%p)\n", $rdi` or others, set empty context sections and just look at the `dprintf` outputs.

**Comment by Aplet123** [comment]
I think it'd be better if we had a setting to enable/disable automatically displaying the context for this case

**Comment by disconnect3d** [comment]
> I think it'd be better if we had a setting to enable/disable automatically displaying the context for this case

So you mean a setting to disable displaying of context at all instead of changing the context sections? Hmm that probably makes sense.

Something like: `context --enable|--disable` or `context enable|disable`?

**Comment by Gaurav23V** [comment]
Hi, I'm new to contributing and would like to work on this "good first issue".

Reading the discussion, it seems there were two main ideas proposed:

1. Make the ctx command display the default context when set context-sections '' is used (the original proposal).
2. Keep the ctx command showing nothing in that case (respecting the empty setting), but add a new setting/command (like set auto-context off or context disable) to control the automatic context display, as suggested by @Aplet123 and considered by @disconnect3d.

Could you please clarify which approach is preferred? I'm happy to help implement whichever solution is decided upon. Thanks!

**Comment by disconnect3d** [comment]
I think that 2) makes sense. cc: @marywinter3
